source/core_http_client.c

```diff
@@ -567,20 +567,30 @@ static uint32_t getZeroTimestampMs( void )
 }
 
 /*-----------------------------------------------------------*/
-
-static int8_t caseInsensitiveStringCmp( const unsigned char * str1,
-                                        const unsigned char * str2,
+static int8_t caseInsensitiveStringCmp( const char * str1,
+                                        const char * str2,
                                         size_t n )
 {
     size_t i = 0U;
-    /* Inclusion of inbetween variables for coverity rule 13.2 compliance */
-    int32_t firstChar;
-    int32_t secondChar;
+    /* Inclusion of inbetween variables for MISRA rule 13.2 compliance */
+    int8_t firstChar;
+    int8_t secondChar;
 
     for( i = 0U; i < n; i++ )
     {
-        firstChar = toupper( ( int32_t ) ( str1[ i ] ) );
-        secondChar = toupper( ( ( int32_t ) str2[ i ] ) );
+        firstChar = ( int8_t ) str1[ i ];
+        secondChar = ( int8_t ) str2[ i ];
+
+        /* Subtract 32 to go from uppercase to lowercase ASCII character */
+        if( ( firstChar >= 97 ) && ( firstChar <= 122 ) )
```

Comment:
can we not use character literals here? 'A' 'Z'. it would probably also eliminate the need to cast to int8_t unless there was some other reason for that.

Reply:
It would seem that you'd be 100% right. I had left them in from doing the previous int32_t related changes, but it looks like swapping to just char's still passes all unit tests and hits all branch coverage.